### DISCUSSION TITLE
what():  [json.exception.parse_error.101] parse error at line 1, column 6: syntax error while parsing value - unexpected ':'; expected end of input

### DISCUSSION THREAD

**Discussion by botoboua** [nlohmann/json#3161] [Q&A] [answered]
```
#include <stdio.h>
#include <sstream>
#include <fstream>
#include "single_include/nlohmann/json.hpp"
using json = nlohmann::json;
using namespace std;

void main(void)
{
        string json_msg = "\"kAy\":[{\"first\":\"second\"}]";
	json jtokens = json::parse(json_msg);
}
```

comple g++ (c++14)
after start program i catch this error - "terminate called after throwing an instance of 'nlohmann::detail::parse_error'
what():  [json.exception.parse_error.101] parse error at line 1, column 6: syntax error while parsing value - unexpected ':'; expected end of input"

how fix it? on windows is work, but on linux have error!

**Comment by nlohmann** [comment, marked as answer]
Your input is invalid JSON. It seems an opening brace for the object is missing. There should also be an error on Windows.

**Reply by nlohmann** [reply]
Do you need further assistance with this issue?

**Comment by anu-n87** [comment]
@nlohmann : I have a similar issue. My json file contains:
{"Anna" : { 
      "age": 18,
      "profession": "student"},
 "Ben" : {
      "age" : "nineteen",
      "profession": "mechanic"}
 }

Below is the code:
        nlohmann::json pointsTest;
	string pointsFTest = "Resources/sample.json";
	ifstream pointsFileTest(pointsFTest);
	cout<<"***PRINTING OUTPUT***"<<pointsFileTest.rdbuf()<<"*********"<<endl;
        pointsFileTest >> pointsTest;

The output has been printed succesfully. However, deserializing gives below error.
terminate called after throwing an instance of 'nlohmann::detail::parse_error'
  what():  [json.exception.parse_error.101] parse error at line 1, column 1: syntax error while parsing value - unexpected end of input; expected '[', '{', or a literal
Aborted (core dumped)

**Reply by nlohmann** [reply]
You try to read the file twice. That does not work.

**Reply by anu-n87** [reply]
@nlohmann : Thanks!
